.github/workflows/status.yaml

```diff
@@ -0,0 +1,64 @@
+name: Custom GH Status from Workflow Artifacts
+
+on:
+  workflow_run:
+    workflows: ["pr"]
+    types:
+      - completed
+
+jobs:
+  process_artifacts:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: gh-status
+          path: downloaded_artifacts/
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Read artifact JSON
+        id: read_artifact
+        run: |
+          CONTENT=$(cat downloaded_artifacts/gh-status.json)
+          echo "content=$CONTENT" >> $GITHUB_OUTPUT
```

Comment:
This doesn't necessarily need to be its own step.

You can save a few lines of code by moving the `CONTENT=$(cat downloaded_artifacts/gh-status.json)` command to the subsequent step.

Reply:
Done 👍